docs/DevGuide/DomainConcepts.md

```diff
@@ -4,61 +4,94 @@ See LICENSE.txt for details.
-  A reference cell embedded into a subset of the computational domain. Blocks
-  are identified by unique integral values.
+  The computational domain is partitioned into a set of non-overlapping,
+  distorted reference cells called Blocks. Each Block must have at most one
+  neighboring Block in each Direction.  The reference cell is embedded into a
+  subset of the computational domain using a Coordinate Map from the
+  logical frame of the Block to the global inertial frame.  Blocks are
+  identified by unique integral values.
+
+* Block Logical Coordinate Frame:<br>
+  The logical coordinate frame of a Block, denoted by Frame::BlockLogical.
+  The only requirement upon the logical coordinate frames of neighboring Blocks
+  is that they have the same Coordinate Map from their logical coordinate frame
+  to the global inertial frame on their shared boundary up to a mapping that
+  swaps or negates the logical coordinate axes.  In other words, at each point
+  on their shared boundary, the logical coordinates of the two blocks are
+  equal after possibly applying a permutation and sign flips.
+
+* Grid Coordinate Frame:<br>
+  For time-dependent coordinate maps, it is useful (e.g. for computational
+  efficiency) to split the Coordinate Map from Frame::BlockLogical to
+  Frame::Inertial into a composition of two maps.  This is done by introducing
+  an intermediate coordinate frame (denoted by Frame::Grid) such that the
+  mapping from Frame::BlockLogical to Frame::Grid is time-independent, and
+  the mapping from Frame::Grid to Frame::Inertial is time-dependent.
 
 * Orientation:<br>
-  The information of how the Logical Coordinates between Blocks are related.
+  The information of how the Block Logical Coordinates of neighboring Blocks are
+  related.
 
 * \ref BlockNeighbor "Block Neighbor":<br>
-  The identity and Orientation of a neighbor of a given Block.
+  The identity and Orientation of a neighboring Block of a given Block.
 
-* Direction:<br>
-  A logical coordinate axis and a label "Upper" or "Lower" depending on whether
-  the Direction is aligned with or anti-aligned with the logical coordinate
-  axis, respectively.
+* Element:<br>
+  A reference cell that is a refined subregion of a Block defined by its
+  Segments in each dimension. The properties of the Element are inherited from
+  its Block, i.e. it is self-similar to the Block.
 
 * Refinement Level:<br>
   The number of times a Block is split in half in a given dimension.
 
 * Segment:<br>
-  The specific subset of the interval \f$[-1, 1]\f$ defined by the Refinement Level and an integer label
-  such that the Segment's interval is \f$[-1 + 2*(N/D), -1 + 2*(N+1)/D]\f$ where N is the integer label and
-  D is \f$2^{(\textrm{Refinement Level})}\f$.
+  In each dimension, the specific subset of the block logical coordinate
+  interval \f$[-1, 1]\f$ defined by the Refinement Level and an integer label
+  such that the Segment's interval is \f$[-1 + 2*(N/D), -1 + 2*(N+1)/D]\f$
```

Comment:
`\f$[-1 + 2 * (N/D), -1 + 2 * (N+1)/D]\f$`

Reply:
no `*` in tex

Reply:
Oh I misread this! Ignore my comment :slightly_smiling_face: